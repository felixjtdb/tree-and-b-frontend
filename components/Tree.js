import React, { Component } from 'react';
import { StyleSheet, Text, View, Image  } from 'react-native';
import { Constants } from 'expo'


export default class Tree extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Image
          style={styles.image}
          resizeMode={"cover"}
          source={{ uri: this.props.imageURL }}
        />
        <Text style={styles.name}> {this.props.name} </Text>
        <Text style={styles.description}> {this.props.description} </Text>
        <Text style={styles.location}> {this.props.location} </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: Constants.statusBarHeight,
    backgroundColor: '#ecf0f1',
  },
  image: {
    height: 150,
    width: 150,
    borderRadius: 150,
  },
  name: {
    color: 'red'
  },
  description: {

  },
  location: {

  }
});
