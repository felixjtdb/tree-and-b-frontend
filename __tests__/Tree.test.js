import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import Tree from '../components/Tree.js'

import renderer from 'react-test-renderer';

const tree = renderer.create(
  <Tree
    name={'test_tree'}
    description={'test_description'}
    imageURL={'test_imageURL'}
    location={'test_location'}
  />
).toJSON();

test('render matches snapshot', () => {
  expect(tree).toMatchSnapshot();
})
