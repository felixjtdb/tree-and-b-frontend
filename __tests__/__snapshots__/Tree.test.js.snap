// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ecf0f1",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "test_imageURL",
      }
    }
    style={
      Object {
        "borderRadius": 150,
        "height": 150,
        "width": 150,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "red",
      }
    }
  >
     
    test_tree
     
  </Text>
  <Text
    style={Object {}}
  >
     
    test_description
     
  </Text>
  <Text
    style={Object {}}
  >
     
    test_location
     
  </Text>
</View>
`;
